.github/workflows/spelling.yml

```diff
@@ -0,0 +1,145 @@
+
+on:
+  push:
+    branches:
+    - "**"
+    tags-ignore:
+    - "**"
+  pull_request_target:
+    branches:
+    - "**"
+    types:
+    - 'opened'
+    - 'reopened'
+    - 'synchronize'
+  issue_comment:
+    types:
+    - 'created'
+
+jobs:
+  spelling:
+    name: Check Spelling
+    permissions:
+      contents: read
+      pull-requests: read
+      actions: read
+      security-events: write
+    outputs:
+      followup: ${{ steps.spelling.outputs.followup }}
+    runs-on: ubuntu-latest
+    if: ${{ contains(github.event_name, 'pull_request') || github.event_name == 'push' }}
+    concurrency:
+      group: spelling-${{ github.event.pull_request.number || github.ref }}
+      # note: If you use only_check_changed_files, you do not want cancel-in-progress
+      cancel-in-progress: true
+    steps:
+    - name: check-spelling
+      id: spelling
+      uses: check-spelling/check-spelling@v0.0.21
+      with:
+        suppress_push_for_open_pull_request: 1
+        checkout: true
+        check_file_names: 1
+        spell_check_this: openrewrite/rewrite-docs@master
+        post_comment: 0
+        use_magic_file: 1
+        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'openrewrite' && 1 }}
+        use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
+        extra_dictionary_limit: 10
+        dictionary_source_prefixes: |
+          {"cspell": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20220816/dictionaries/", "cspell1": "https://raw.githubusercontent.com/check-spelling/cspell-dicts/v20230509/dictionaries/"}
+        check_extra_dictionaries: ''
+        extra_dictionaries:
+          cspell1:software-terms/dict/softwareTerms.txt
+          cspell1:aws/aws.txt
+          cspell1:npm/dict/npm.txt
+          cspell1:java/src/java.txt
+          cspell1:k8s/dict/k8s.txt
+          cspell1:typescript/dict/typescript.txt
+          cspell1:filetypes/filetypes.txt
+          cspell1:fullstack/dict/fullstack.txt
+          cspell1:python/src/python/python-lib.txt
+          cspell1:scala/dict/scala.txt
+          cspell1:php/dict/php.txt
+          cspell1:node/dict/node.txt
+
+  update:
+    name: Update PR
+    permissions:
+      contents: write
+      pull-requests: write
+      actions: read
+    runs-on: ubuntu-latest
+    if: ${{
+        github.repository_owner != 'openrewrite' &&
+        github.event_name == 'issue_comment' &&
+        github.event.issue.pull_request &&
+        contains(github.event.comment.body, '@check-spelling-bot apply')
+      }}
+    concurrency:
+      group: spelling-update-${{ github.event.issue.number }}
+      cancel-in-progress: false
+    steps:
+    - name: apply spelling updates
+      uses: check-spelling/check-spelling@v0.0.21
+      with:
+        experimental_apply_changes_via_bot: 1
+        checkout: true
+        ssh_key: "${{ secrets.CHECK_SPELLING }}"
```

Comment:
So if I understand correctly this part is only if we wish to update some of these files through a fork, right?

Reply:
Right.

To trigger it:
1. Create your own personal fork
2. In the fork, create a branch with a typo (or an unknown word...)
3. Create a pull request into the default branch of the fork.
4. Go to the ❌ for the status and follow the instructions. The first round through, it'll walk you through creating a deploy key (which is optional) -- if you do that, the second time you use the feature, it will be a bit smoother.